libcudacxx/include/cuda/__iterator/transform_iterator.h

```diff
@@ -0,0 +1,501 @@
+    noexcept(noexcept(_CUDA_VSTD::invoke(const_cast<_Fn&>(*__func_), *__current_)))
+  {
+    return _CUDA_VSTD::invoke(const_cast<_Fn&>(*__func_), *__current_);
+  }
+
+  //! @brief Increments the stored iterator
+  _LIBCUDACXX_HIDE_FROM_ABI constexpr transform_iterator& operator++() noexcept(noexcept(++__current_))
+  {
+    ++__current_;
+    return *this;
+  }
+
+  //! @brief Increments the stored iterator
+  _LIBCUDACXX_HIDE_FROM_ABI constexpr auto operator++(int) noexcept(noexcept(++__current_))
+  {
+    if constexpr (_CUDA_VSTD::forward_iterator<_Iter>)
+    {
+      auto __tmp = *this;
+      ++*this;
+      return __tmp;
+    }
+    else
+    {
+      ++__current_;
+    }
+  }
+
+  //! @brief Decrements the stored iterator
+  _CCCL_TEMPLATE(class _Iter2 = _Iter)
+  _CCCL_REQUIRES(_CUDA_VSTD::bidirectional_iterator<_Iter2>)
+  _LIBCUDACXX_HIDE_FROM_ABI constexpr transform_iterator& operator--() noexcept(noexcept(--__current_))
+  {
+    --__current_;
+    return *this;
+  }
+
+  //! @brief Decrements the stored iterator
+  _CCCL_TEMPLATE(class _Iter2 = _Iter)
+  _CCCL_REQUIRES(_CUDA_VSTD::bidirectional_iterator<_Iter2>)
+  _LIBCUDACXX_HIDE_FROM_ABI constexpr transform_iterator
+  operator--(int) noexcept(_CUDA_VSTD::is_nothrow_copy_constructible_v<_Iter> && noexcept(--__current_))
+  {
+    auto __tmp = *this;
+    --*this;
+    return __tmp;
+  }
+
+  //! @brief Advances this \c transform_iterator by \p __n
+  //! @param __n The number of elements to advance
+  _CCCL_TEMPLATE(class _Iter2 = _Iter)
+  _CCCL_REQUIRES(_CUDA_VSTD::random_access_iterator<_Iter2>)
+  _LIBCUDACXX_HIDE_FROM_ABI constexpr transform_iterator&
+  operator+=(difference_type __n) noexcept(noexcept(__current_ += __n))
+  {
+    __current_ += __n;
+    return *this;
+  }
+
+  //! @brief Decrements this \c transform_iterator by \p __n
+  //! @param __n The number of elements to decrement
+  _CCCL_TEMPLATE(class _Iter2 = _Iter)
+  _CCCL_REQUIRES(_CUDA_VSTD::random_access_iterator<_Iter2>)
+  _LIBCUDACXX_HIDE_FROM_ABI constexpr transform_iterator&
+  operator-=(difference_type __n) noexcept(noexcept(__current_ -= __n))
+  {
+    __current_ -= __n;
+    return *this;
+  }
+
+  //! @brief Subscripts the stored iterator by \p __n and applies the stored functor to the result
+  //! @param __n The additional offset
+  //! @returns _CUDA_VSTD::invoke(__func_, __current_[__n])
+  _CCCL_TEMPLATE(class _Iter2 = _Iter)
+  _CCCL_REQUIRES(_CUDA_VSTD::random_access_iterator<_Iter2>)
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr decltype(auto) operator[](difference_type __n) const
+    noexcept(__transform_iterator_nothrow_subscript<_Fn, _Iter2>)
+  {
+    return _CUDA_VSTD::invoke(const_cast<_Fn&>(*__func_), __current_[__n]);
+  }
+
+  //! @brief Compares two \c transform_iterator \p __lhs and \p __rhs for equality
+  //! @param __lhs The left \c transform_iterator
+  //! @param __rhs The right \c transform_iterator
+  //! @return true if the stored iterators compares equal
```

Comment:
Q: Does this comment really add any value?

Suggestion: remove it.

Applies to more comments below.

Reply:
Actually that is a *really* good question, because implementation of `transform_view` does not consider the functor to be part of the equation, which is ... strange and probably a LWG issue

Reply:
Then maybe simplify:
```suggestion
  //! @brief Compares two \c transform_iterator for equality, ignoring their functors
```